src/index.js

```diff
@@ -0,0 +1,96 @@
+const path = require('path')
+    , checker = require('license-checker')
+    , satisfies = require('spdx-satisfies')
+    , licenseList = require('spdx-license-list/full');
+
+function fulfill(config = {}) {
+  return new Promise((resolve, reject) => {
+    const { overrides, agreedLicenses, startPath, acceptableLicenseFiles, ignorePackages = [] } = config;
+    const defaultLicenseTexts = licenseList;
+    checker.init({ customFormat: {}, start: startPath }, function(err, packages) {
+      if (err) {
+        reject(err);
+        return;
+      }
+
+      let licenseFulfillments = {};
+
+      Object.keys(packages).forEach(key => {
+        const curPackage = packages[key];
+
+        Object.assign(curPackage, overrides[key]);
+
+        if (typeof curPackage.licenses !== 'string') {
+          curPackage.agreedLicense = null;
+          return;
+        }
+
+        if (! curPackage.forceAccept) {
+          for (const agreedLicense of agreedLicenses) {
+            try {
+              if (satisfies(agreedLicense, curPackage.licenses)) {
+                curPackage.agreedLicense = agreedLicense;
+                break;
+              }
+            } catch(e) {
+              curPackage.agreedLicense = null;
+            }
+          }
+
+          if (! curPackage.agreedLicense) {
```

Comment:
It's possible when no agreed licenses satisfy the package license _and_ none of them throw (it only throws when non-SPDX licenses are used).